src/assets-importer/index.js

```diff
@@ -0,0 +1,129 @@
+    },
+  }).then(res => res.json());
+}
+
+function normalizeFilename({ name, regex }) {
+  const matches = name.match(regex);
+
+  return matches ? matches.pop() : null;
+}
+
+async function download({
+  url,
+  filename,
+  dist = '',
+}) {
+  const distPath = `${process.cwd()}${dist}`;
+  await fs.mkdir(distPath, { recursive: true }, (err) => {
+    if (err) throw err;
+  });
+  const writer = fs.createWriteStream(`${distPath}/${filename}`);
+
+  try {
+    const response = await fetch(url);
+    response.body.pipe(writer);
+
+    return new Promise((resolve, reject) => {
+      writer.on('finish', resolve(filename));
+      writer.on('error', reject);
+    });
+  } catch(err) {
+    return Promise.reject(new Error(err));
+  }
+}
+
+async function importAssetsFromFigma({
+  fileKey,
+  format,
+  dist,
+  filenameRegex,
+  shouldMatchRegex,
+}) {
+  const file = await fetchFromFigma(`/files/${fileKey}/components`);
+
+  if (!file.meta.components || file.meta.components.length === 0) {
+    console.warn('No components found for file', fileKey);
+    return;
+  }
+
+  const filenameMap = file.meta.components
+    .reduce((map, component) => {
+      const isMatch = Array.isArray(component.name.match(shouldMatchRegex));
+      if (isMatch) {
+        const filename = normalizeFilename({ name: component.name, regex: filenameRegex });
+        map[component.node_id] = filename;
+      }
+
+      return map;
+    }, {});
+
+  const ids = Object.keys(filenameMap).join(',');
+  const assets = await fetchFromFigma(`/images/${fileKey}?ids=${ids}&format=${format}`);
+
+  if (assets.images === undefined) {
+    console.warn('No images found for regexp', shouldMatchRegex);
+    return;
+  }
+
+  const downloadedFiles = await Promise.all(
+    Object.entries(assets.images)
+      .map(([id, url]) => url && download({
+        url,
+        filename: `${filenameMap[id]}.${format}`,
+        dist,
+      })),
+  );
+
+  createIndex({
+    files: downloadedFiles,
+    dist,
+    exportNameRegex: new RegExp(`ic_(.*).${format}`),
+    suffix: 'Icon',
+  });
+}
+
+function createIndex({ files, dist, exportNameRegex = '*', suffix = '' }) {
+  const fileContent = [...new Set(files)]
+    .reduce((content, file) => {
+      const nameMatches = file.match(exportNameRegex);
+      if (!nameMatches) {
+        return content;
+      }
+
+      const exportName = _camelCase(nameMatches.pop()) + suffix;
+      return `export { default as ${exportName} } from './${file}';\n${content}`;
+    }, '');
+
+  fs.writeFile(`${process.cwd()}${dist}/index.js`, fileContent, (err) => {
+    if (err) console.error('Failed to create index', err);
+  });
+}
+
+
+function importAll() {
+  CONFIG.FIGMA_FILES.forEach(FIGMA_FILE => importAssetsFromFigma({
+    fileKey: FIGMA_FILE.KEY,
+    format: FIGMA_FILE.FORMAT,
+    filenameRegex: FIGMA_FILE.FILENAME_REGEX,
+    shouldMatchRegex: FIGMA_FILE.SHOULD_MATCH_REGEX,
+    dist: FIGMA_FILE.DIST,
+  }));
+};
+
+importAll();
```

Comment:
You could move the call to the top, so as soon as we open the file, we see the main part (without scroll). I usually, when I use function definition, put main part on top (or exposed exposed members - is something I've picked years ago working on [Angular 1](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#accessible-members-up-top)  actually :D ) and then who's/when interested you can scroll down to implementation details. 

However, we often assign function to variables (with `const` often being the preferred way) and due to hoisting we have to reverse the order. If it's like that due consistency, then all good also :+1:  

Since mentioned already, what are you preferred ways @homeday-de/frontend-engineering ?

Reply:
Ah I see your reference, because use used "main" in your comment, C and Python came directly to my mind :D

I usually declare then use (as this rule forced me to do: [no-use-before-define](https://eslint.org/docs/rules/no-use-before-define) :D), but I see your point, it's better to know what's happening from the beginning.

This is something that bothered me a bit (the call inside the file), so I'll change it to the command line, I'll keep the file just for the definitions, and invoke the `importAll` (and maybe also add an `importFileByName` function as well, to avoid importing everything everytime) from the command.

Thanks for bring up this topic, it confirms my concerns :D

Reply:
Ah, yes...no use before define. :) It's good to reason and have this discussions to see different approaches. :+1:   We can also keep it, I think it's a matter of preference. 

On John Papa (the Angular reference) note - he's really cool, has usually some great tips (even old angular guide gives some good reasoning for code structuring outside angular). :ok_hand:

Reply:
I went with something similar, please check the new commits :)